packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -319,14 +319,14 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
 ///  * [showBottomSheet] and [ScaffoldState.showBottomSheet], for showing
 ///    non-modal bottom sheets.
 ///  * <https://material.google.com/components/bottom-sheets.html#bottom-sheets-modal-bottom-sheets>
-Future<T> showModalBottomSheet<T>({
+Future<void> showModalBottomSheet({
   @required BuildContext context,
   @required WidgetBuilder builder,
 }) {
   assert(context != null);
   assert(builder != null);
   assert(debugCheckHasMaterialLocalizations(context));
-  return Navigator.push(context, _ModalBottomSheetRoute<T>(
+  return Navigator.push(context, _ModalBottomSheetRoute(
```

Comment:
Can't this potentially return something other than a `Future<void>`?

Reply:
I haven't found any place where something other than `void` (or `Null`) is returned. The only place where `T` is used is to create a `Completer<T>` which is only [complete without value](https://github.com/flutter/flutter/pull/22899/files#diff-ad7e5ee7be5375e4242652ddc6448da7R1245). So I looks fine to remove the type parameter on `_ModalBottomSheetRoute` and make it extending `PopupRoute<void>`.

Reply:
i would expect `Navigator.pop()` to have its value used here.